src/Worker.php

```diff
@@ -84,18 +84,18 @@ protected function tick($queue)
             return false;
         }
 
-        $message = $this->driver->dequeue($queue);
-        if (empty($message) && $this->drain) {
+        list($command, $job) = $this->driver->dequeue($queue);
+        if (empty($command) && $this->drain) {
             $this->event->drained();
             return false;
-        } elseif (empty($message)) {
+        } elseif (empty($command)) {
             return true;
         }
 
-        $command = unserialize($message);
```

Comment:
This is the responsibility of the driver, as the serialization already happens there.